manim/mobject/svg/text_mobject.py

```diff
@@ -1056,11 +1058,510 @@ def text2svg(self):
+
+            colormap.append(
+                {
+                    "start": start,
+                    "end": end,
+                    "color": tag.group(1),
+                    "start_offset": start_offset,
+                    "end_offset": end_offset,
+                }
+            )
+        self.text = re.sub("<color[^>]+>(.+?)</color>", r"\1", self.text, 0, re.S)
+        return colormap
+
+    def __repr__(self):
+        return f"MarkupText({repr(self.original_text)})"
+
+    def gen_chars(self):
+        chars = VGroup()
+        submobjects_char_index = 0
+        for char_index in range(self.text.__len__()):
+            if (
+                self.text[char_index] == " "
+                or self.text[char_index] == "\t"
+                or self.text[char_index] == "\n"
+            ):
+                space = Dot(radius=0, fill_opacity=0, stroke_opacity=0)
+                if char_index == 0:
+                    space.move_to(self.submobjects[submobjects_char_index].get_center())
+                else:
+                    space.move_to(
+                        self.submobjects[submobjects_char_index - 1].get_center()
+                    )
+                chars.add(space)
+            else:
+                chars.add(self.submobjects[submobjects_char_index])
+                submobjects_char_index += 1
+        return chars
+
+    def str2style(self, string):
+        """Internally used function. Converts text to Pango Understandable Styles."""
+        if string == NORMAL:
+            return pangocffi.Style.NORMAL
+        elif string == ITALIC:
+            return pangocffi.Style.ITALIC
+        elif string == OBLIQUE:
+            return pangocffi.Style.OBLIQUE
+        else:
+            raise AttributeError("There is no Style Called %s" % string)
+
+    def str2weight(self, string):
+        """Internally used function. Convert text to Pango Understandable Weight"""
+        if string == NORMAL:
+            return pangocffi.Weight.NORMAL
+        elif string == BOLD:
+            return pangocffi.Weight.BOLD
+        elif string == THIN:
+            return pangocffi.Weight.THIN
+        elif string == ULTRALIGHT:
+            return pangocffi.Weight.ULTRALIGHT
+        elif string == LIGHT:
+            return pangocffi.Weight.LIGHT
+        elif string == SEMILIGHT:
+            return pangocffi.Weight.SEMILIGHT
+        elif string == BOOK:
+            return pangocffi.Weight.BOOK
+        elif string == MEDIUM:
+            return pangocffi.Weight.MEDIUM
+        elif string == SEMIBOLD:
+            return pangocffi.Weight.SEMIBOLD
+        elif string == ULTRABOLD:
+            return pangocffi.Weight.ULTRABOLD
+        elif string == HEAVY:
+            return pangocffi.Weight.HEAVY
+        elif string == ULTRAHEAVY:
+            return pangocffi.Weight.ULTRAHEAVY
+        else:
+            raise AttributeError("There is no Font Weight Called %s" % string)
```

Comment:
Suggestion: It is just a copy of the things is `Text`. Rather than doing that, define it as a static method there and that can be used here.

Reply:
My suggestion would be to define those like `remove_invisible_chars` outside of any class. In the long run, this class might replace the classical `Text` class (one never knows) and have us "move" the methods from one to another. What do you think?

Reply:
Better inherit from `Text` class and use these methods after `__init__` it? What do you think?

Reply:
It used to inherit from `Text`, I changed it to `SVGMobject` as per your earlier comment. I have another suggestion: those two methods actually do not logically belong to `Text`. We could have a `PangoUtils` class with these two methods as static and both `Text` and `MarkupText` invoke them from there. Would that be OK for you?

Reply:
Looks good to me.